bundles/org.openhab.binding.opensprinkler/src/main/java/org/openhab/binding/opensprinkler/internal/handler/OpenSprinklerDeviceHandler.java

```diff
@@ -68,21 +68,16 @@ protected void updateChannel(ChannelUID channel) {
     }
 
     @Override
-    public void handleCommand(ChannelUID channelUID, Command command) {
-        // nothing to do here
-    }
-
-    @Override
-    public void updateChannels() {
+    public void initialize() {
         ChannelUID currentDraw = new ChannelUID(thing.getUID(), "currentDraw");
         ThingBuilder thingBuilder = editThing();
-        Channel currentDrawChannel = ChannelBuilder.create(currentDraw, "Number:ElectricCurrent")
-                .withType(new ChannelTypeUID(BINDING_ID, SENSOR_CURRENT_DRAW)).withLabel("Current Draw")
-                .withDescription("Provides the current draw.").build();
         try {
             getApi().currentDraw();
 
             if (thing.getChannel(currentDraw) == null) {
```

Comment:
Since you never remove the channel, you could move this to the most upper level.

```
if (thing.getChannel == null) {
   try {
       getApi.currentDraw();

       ThingBuilder thingBuilder = editThing();
       Channel currentDrawChannel = ...
       thingBuilder.withChannel(currentDrawChannel);
       updateThing(thingBuilder.build());
   } catch (...) {
      logger.debug(...);
   }
}
```

Reply:
Done